src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -40149,6 +40151,579 @@ public void CopyTupleUnconstrainedElementNullability_02()
+    {
+        int? y = (int?)x;
+        if (x == null) return;
+        int? z = (int?)x;
+        x.Value.ToString();
+        y.Value.ToString(); // 1
+        z.Value.ToString();
+    }
+}";
+            var comp = CreateCompilation(source, options: WithNonNullTypesTrue());
+            comp.VerifyDiagnostics(
+                // (9,9): warning CS8629: Nullable value type may be null.
+                //         y.Value.ToString(); // 1
+                Diagnostic(ErrorCode.WRN_NullableValueTypeMayBeNull, "y.Value").WithLocation(9, 9));
+        }
+
+        [Fact]
+        [WorkItem(32531, "https://github.com/dotnet/roslyn/issues/32531")]
+        public void Conversions_ExplicitNullable_05()
+        {
+            var source =
+@"class Program
+{
+    static void F1(int? x1)
+    {
+        int y1 = (int)x1; // 1
+    }
+    static void F2(int? x2)
+    {
+        if (x2 == null) return;
+        int y2 = (int)x2;
+    }
+    static void F3(int? x3)
+    {
+        long y3 = (long)x3; // 2
+    }
+    static void F4(int? x4)
+    {
+        if (x4 == null) return;
+        long y4 = (long)x4;
+    }
+}";
+            var comp = CreateCompilation(source, options: WithNonNullTypesTrue());
+            comp.VerifyDiagnostics(
+                // (5,18): warning CS8629: Nullable value type may be null.
+                //         int y1 = (int)x1; // 1
+                Diagnostic(ErrorCode.WRN_NullableValueTypeMayBeNull, "(int)x1").WithLocation(5, 18),
+                // (14,19): warning CS8629: Nullable value type may be null.
+                //         long y3 = (long)x3; // 2
+                Diagnostic(ErrorCode.WRN_NullableValueTypeMayBeNull, "(long)x3").WithLocation(14, 19));
+        }
+
+        [Fact]
+        [WorkItem(32531, "https://github.com/dotnet/roslyn/issues/32531")]
+        public void Tuple_Conversions_ImplicitNullable_01()
+        {
+            var source =
+@"struct S
+{
+    internal object? F;
+}
+class Program
+{
+    static void F()
+    {
+        S x = new S();
+        S y = new S() { F = 1 };
+        (S, S) t = (x, y);
+        (S? x, S? y) u = t;
+        (S?, S?) v = (x, y);
+        t.Item1.F.ToString(); // 1
+        t.Item2.F.ToString();
+        u.x.Value.F.ToString(); // 2
+        u.y.Value.F.ToString();
+        v.Item1.Value.F.ToString(); // 3
+        v.Item2.Value.F.ToString();
+    }
+}";
+            var comp = CreateCompilation(source, options: WithNonNullTypesTrue());
+            // https://github.com/dotnet/roslyn/issues/32599: Support implicit tuple conversions
+            // (should report warnings for // 1, 2, 3 only).
+            comp.VerifyDiagnostics(
+                // (14,9): warning CS8602: Possible dereference of a null reference.
+                //         t.Item1.F.ToString(); // 1
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "t.Item1.F").WithLocation(14, 9),
+                // (16,9): warning CS8629: Nullable value type may be null.
```

Comment:
>// (16,9): warning CS8629: Nullable value type may be null. [](start = 16, length = 59)

It looks like this warning is unexpected, but the comment above sounds like it is because it is on line //2

Reply:
`// 2` is intended to represent the dereference of `F`.

---
In reply to: [251103126](https://github.com/dotnet/roslyn/pull/32729#discussion_r251103126) [](ancestors = 251103126)

Reply:
> // 2 is intended to represent the dereference of F.

How one is supposed to infer that from "(should report warnings for // 1, 2, 3 only)"?

---
In reply to: [251104699](https://github.com/dotnet/roslyn/pull/32729#discussion_r251104699) [](ancestors = 251104699,251103126)